### PR TITLE
Updated Sphinx developer docs with OS X notes.

### PR DESCRIPTION
**Pull request by liangfok**

### CODE REVIEW

File: drake/doc/development_on_osx.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    pip install hg+http://bitbucket.org/pygame/pygame
+
+Ensure your ``PYTHONPATH`` environment variable includes Drake's site-package and dist-packages directories:
+
+    export PYTHONPATH=$PYTHONPATH:[DRAKE_ROOT]/build/lib/python2.7/site-packages:[DRAKE_ROOT]/build/lib/python2.7/dist-packages
+
+
+
+Running gitk
+============
```
Comment (RussTedrake) [inline drake/doc/development_on_osx.rst]: i personally use sourcetree https://www.sourcetreeapp.com/

Comment (liangfok) [inline drake/doc/development_on_osx.rst]: I downloaded / installed Source Tree yesterday but haven't used it much yet. Will update this page with Source Tree notes once I get more experience with it.

Comment (RussTedrake) [inline drake/doc/development_on_osx.rst]: the notes as written sound like you're listing these as the recommended tools.   perhaps prepend some text which just indicates that these have been useful to some?

Comment (liangfok) [inline drake/doc/development_on_osx.rst]: Oops!  I've updated this PR to include such a statement. PTAL.

**Pull request merged**